docs/pnp/migrate-vsto-to-office-web-add-in-shared-code-library-tutorial.md

```diff
@@ -0,0 +1,427 @@
+
+![Diagram of VSTO Add-in and Office web add-in using a shared code library](../images/vsto-migration-shared-code-library.png)
+
+Skills and techniques in this guide:
+
+- Create a shared class library by refactoring code into a .NET class library.
+- Create a REST API wrapper using ASP.NET Core for the shared class library.
+- Call the REST API from the Office web add-in to access shared code.
+
+## Prerequisites
+
+To set up your development environment:
+
+1. Install [Visual Studio 2019](https://visualstudio.microsoft.com/downloads/).
+2. Install the following workloads:
+    a. ASP.NET and web development
+    b. .NET Core cross-platform development
+    c. Office/SharePoint development
+    d. Visual Studio Tools for Office (VSTO) Note that this is an Individual component.
+
+You will also need the following:
+
+- An Office 365 account. You can join the [Office 365 Developer Program](https://aka.ms/devprogramsignup) that includes a free 1 year subscription to Office 365.
+- A Microsoft Azure Tenant. This add-in requires Azure Active Directiory (AD). Azure AD provides identity services that applications use for authentication and authorization. A trial subscription can be acquired here: [Microsoft Azure](https://account.windowsazure.com/SignUp).
+
+## The Cell analyzer VSTO Add-in
+
+This guide uses the [VSTO Add-in shared library for Office web add-in](https://github.com/OfficeDev/PnP-OfficeAddins/tree/vstoshared/Samples/VSTO-shared-code-start) PnP solution. The **/start** folder contains the VSTO Add-in solution that you will migrate. Your goal is to migrate the VSTO Add-in to a modern Office web add-in by sharing code when possible.
+
+> [!NOTE]
+> The sample uses C# but you can apply the techniques in this guide to a VSTO Add-in written in any .NET language.
+
+1. Download the [VSTO Add-in shared library for Office web add-in](https://github.com/OfficeDev/PnP-OfficeAddins/tree/vstoshared/Samples/VSTO-shared-code-start) PnP solution to a working folder on your computer.
+2. Start Visual Studio 2019 and open the **/start/Cell-Analyzer.sln** solution.
+3. From the **Debug** menu choose **Start Debugging**.
+
+The add-in is a custom task pane for Excel. You can select any cell with text, and then choose the **Show Unicode** button. The add-in will display a list of each character in the text along with its corresponding Unicode number.
+
+![Screenshot of the Cell analyzer VSTO add-in running in Excel](../images/pnp-cell-analyzer-vsto-add-in.png)
+
+## Analyze types of code in the VSTO Add-in
+
+The first technique to apply is to analyze the add-in for which parts of code can be shared. In general there are three types of code to deal with when migrating.
```

Comment:
```suggestion
The first technique to apply is to analyze the add-in for which parts of code can be shared. In general, there are three types of code you'll use when migrating.
```
"to deal with" is vernacular and doesn't translate well. You could use "you encounter".

Reply:
fixed